File: datamodel/src/main/java/datamodel/Evaluation.java
package datamodel;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class Evaluation {

    @NotNull
    public abstract EvaluationResult result();

    public abstract boolean recoverable();

    @NotNull
    public abstract Set<String> inclusionMolecularEvents();

    @NotNull
    public abstract Set<String> exclusionMolecularEvents();

    @NotNull
    public abstract Set<String> passSpecificMessages();

    @NotNull
    public abstract Set<String> passGeneralMessages();

    @NotNull
    public abstract Set<String> warnSpecificMessages();

    @NotNull
    public abstract Set<String> warnGeneralMessages();

    @NotNull
    public abstract Set<String> undeterminedSpecificMessages();

    @NotNull
    public abstract Set<String> undeterminedGeneralMessages();

    @NotNull
    public abstract Set<String> failSpecificMessages();

    @NotNull
    public abstract Set<String> failGeneralMessages();
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/Evaluation.kt
package com.hartwig.actin

data class Evaluation(
    val result: EvaluationResult,
    val recoverable: Boolean,
    val inclusionMolecularEvents: Set<String> = emptySet(),
    val exclusionMolecularEvents: Set<String> = emptySet(),
    val passMessages: Set<String> = emptySet(),
    val warnMessages: Set<String> = emptySet(),
    val undeterminedMessages: Set<String> = emptySet(),
    val failMessages: Set<String> = emptySet(),
)
